### ISSUE TITLE
Question about useBufferedData hook

### ISSUE THREAD

**Issue by arinaldi** [samselikoff/2020-07-16-show-new-tweets#1]
Hello Sam,

I really enjoyed your video that accompanies this code base. You're one of the few content creators showing the power and flexibility of SWR.

My question is about the `stale` property returned from the `useBufferedData` hook. Wouldn't that boolean only be accurate if `data` is a primitive type? If `data` is an object (or an array), I think `stale` would always be `true` since two objects (or arrays) with identical contents are not equal.

Thanks,
Tony

**Comment by samselikoff** [comment]
Thanks for the kind words Tony!

In my testing it worked when `data` was an object – in fact you can pull down the code and try it yourself. You'll see the data returned from the network is an object (something like `{ tweets: [ { id: 1, text: 'Lorem ipsum' }, ... ] }` and when I call `setBuffer(data)`, I believe React will return `buffer` in the next render as a variable that points to the exactly same object as `data`. 

In other words,

```js
let [ data, setData ] = useState({ some: 'object' })
let [ buffer, setBuffer ] = useState(data)

data === buffer // true
```

[I just tried this out in a Sandbox](https://codesandbox.io/s/2020-07-16-show-new-tweets-issue-1-06tpw?file=/src/App.js) and it does indeed work this way. So, both variables point to the same object, and therefore the equivalence check `===` works correctly!

**Comment by arinaldi** [comment]
Thanks for the clarification, and keep up the good work.
